### PR TITLE
Feature/lp 109 ProfileViewModel

### PR DESCRIPTION
**Pull request by JoseGR98**
Profile - ViewModel
[LP-109](https://devaptivist.atlassian.net/browse/LP-109)
View Model implementation for the Profile View.
Files added/edited:
- ProfileView that contains the UI
- ProfileViewModel that handles the logic of data
- ProfileModel that contains the data class for the User Profile Data
NavGraph of Profile edited because of new parameter -  TODO

### CODE REVIEW

File: core/data/src/main/java/com/iteneum/ProfileModel.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.iteneum
+
+data class ProfileModel (
```
Comment (irv205) [inline core/data/src/main/java/com/iteneum/ProfileModel.kt]: Rename "ProfileModel" to "Profile"

File: core/navigation/src/main/java/com/iteneum/navigation/graphs/DashBoardNavGraph.kt [hunk 2]
```diff
@@ -14,7 +14,7 @@ fun NavGraphBuilder.dashboardNavGraph(navController: NavHostController) {
         startDestination = DashBoardScreen.Profile.route
     ) {
         composable(route = DashBoardScreen.Profile.route) {
-            ProfileView()
+            ProfileView(navigation = { /*TODO*/ })
```
Comment (irv205) [inline core/navigation/src/main/java/com/iteneum/navigation/graphs/DashBoardNavGraph.kt]: Explain in that TODO what you need to do in that specific part.

File: feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt [hunk 3]
```diff
@@ -28,18 +29,30 @@ import androidx.compose.ui.res.painterResource
...
 import com.iteneum.designsystem.components.LpOutlinedButton
 import com.iteneum.designsystem.theme.LeasePertTheme
 import com.iteneum.designsystem.R as R_DS
 import com.iteneum.profile.R
+import com.iteneum.profile.data.ProfileViewModel
 
 @Composable
 fun ProfileView(
-    /*TODO expected parameters as navigation and view model*/
+    navigation: () -> Unit /* TODO - ProfileView - Check navigation parameter*/ ,
```
Comment (irv205) [inline feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt]: rename this parameter, it depends what do you want, if it's to navigate back you can rename it as "BackToDashBoard" or if it's to navigate to other view you can rename "navigateToNameView"

File: core/data/src/main/java/com/iteneum/ProfileModel.kt (hunk 1 above)
Comment (JoseGR98) [reply core/data/src/main/java/com/iteneum/ProfileModel.kt]: Sure, no problem. Done!

File: core/navigation/src/main/java/com/iteneum/navigation/graphs/DashBoardNavGraph.kt (hunk 2 above)
Comment (JoseGR98) [reply core/navigation/src/main/java/com/iteneum/navigation/graphs/DashBoardNavGraph.kt]: Got it! Done!

File: feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt (hunk 3 above)
Comment (JoseGR98) [reply feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt]: It is for going back, by pressing the arrow component, but not implemented yet. Changed the name for the one you mentioned. Thank you!

**Review by irv205 — APPROVED** [review]

File: feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+/**
+ * File that contains [ProfileViewModel] that works as ViewModel for the Profile View.
+ * It will handle the data work & process when loading & be done with the view.
+ *
+ * @author Jose Miguel Garcia Reyes
+ */
+
+@HiltViewModel
+class ProfileViewModel @Inject constructor(): ViewModel() {
+    private val _profileModelMutable : MutableState<Profile> = mutableStateOf(Profile())
```
Comment (carlosph816) [inline feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: we can change this in the same way you did with profileModel

File: feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt [hunk 5]
```diff
@@ -28,18 +29,30 @@ import androidx.compose.ui.res.painterResource
...
-    val sizes = LeasePertTheme.sizes
+    LaunchedEffect(true) {
+        profileViewModel.getInformation()
+    }
+    ProfileContainer()
+}
 
+@Composable
+fun ProfileContainer(
+    profileViewModel: ProfileViewModel = hiltViewModel()
```
Comment (carlosph816) [inline feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt]: we don't need to give the viewmodel here, you need to implement the hight order functions for events

File: feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt (hunk 4 above)
Comment (JoseGR98) [reply feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: Ok, done!

File: feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt (hunk 5 above)
Comment (JoseGR98) [reply feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt]: Got it, done!

**Comment by JoseGR98** [comment]
New PR made. Code review pending

File: feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                profileModelMutable = Profile(
+                    profileName = "Error",
+                    profileApartment = "Error",
+                    profilePhoneNumber = "1111111111",
+                    profileEmail = "Error",
+                    profileAddress = "Error"
+                )
+            }
+            is DataState.Loading -> {
+                profileModelMutable = Profile(
```
Comment (carlosph816) [inline feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: this shouldn't be here, let's delete and put a "TODO"

File: feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    var profileModelMutable: Profile by mutableStateOf(Profile())
+        private set
+
+    fun getInformation() = viewModelScope.launch {
+        when (response) {
+            is DataState.Success -> {
+                profileModelMutable = response.data
+            }
+            is DataState.Error -> {
+                profileModelMutable = Profile(
```
Comment (carlosph816) [inline feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: this shouldn't be here, let's delete and put a "TODO"

Comment (JoseGR98) [reply feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: Got it. Modified!

File: feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt (hunk 6 above)
Comment (JoseGR98) [reply feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: Got it. Modified!

**Comment by JoseGR98** [comment]
1 code reviewer approved. Waiting for one more approval from you @carlosph816 
Both approved! Only merge left! :D

**Review by carlosph816 — APPROVED** [review]

**Pull request merged**